core/src/main/java/org/infinispan/metrics/impl/ApplicationMetricsRegistry.java

```diff
@@ -34,19 +38,33 @@
  */
 @Scope(Scopes.GLOBAL)
 @SurvivesRestarts
-public final class ApplicationMetricsRegistry {
+public class ApplicationMetricsRegistry {
 
-   private final MetricRegistry applicationMetricRegistry = MetricRegistries.get(MetricRegistry.Type.APPLICATION);
+   private static final Log log = LogFactory.getLog(ApplicationMetricsRegistry.class);
+
+   private final MetricRegistry applicationMetricRegistry;
 
    public ApplicationMetricsRegistry() {
+      applicationMetricRegistry = makeRegistry();
+   }
+
+   protected MetricRegistry makeRegistry() {
+      return MetricRegistries.get(MetricRegistry.Type.APPLICATION);
    }
 
-   public MetricRegistry getRegistry() {
+   public final MetricRegistry getRegistry() {
       return applicationMetricRegistry;
    }
 
    public void register(ResourceDMBean resourceDMBean) {
+      MetricRegistry registry = getRegistry();
+
       ObjectName objectName = resourceDMBean.getObjectName();
+      int metricCounter = 0;
+      if (log.isDebugEnabled()) {
+         log.debugf("Metric registry @%x contains %d metrics. Registering metrics for ObjectName \"%s\" ...",
```

Comment:
IMO this is too low-level even for debug, it and the other messages you added should be at the trace level.

Reply:
done